src/unread/unreadModel.js

```diff
@@ -203,6 +204,47 @@ function streamsReducer(
       return deleteMessages(state, action.messages);
     }
 
+    case EVENT_UPDATE_MESSAGE: {
+      const { stream_id } = action;
+      if (stream_id == null) {
+        // Not stream messages, or else a pure content edit (no stream/topic change.)
+        //
+        // As of this writing, the docs actually promise this field for all
+        // updates to stream messages.  Empirically it's absent on pure
+        // content edits, but present on edits affecting either the stream
+        // or topic:
+        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60update_message.60.20event/near/1296823
+        return state;
+      }
+
+      if (
+        (action.subject === action.orig_subject || action.orig_subject == null)
+        && (action.new_stream_id === stream_id || action.new_stream_id == null)
+      ) {
```

Comment:
I think a check like this makes a lot of sense, given that we don't have confirmation on what we've observed empirically; see above:

```
        // As of this writing, the docs actually promise this field for all
        // updates to stream messages.  Empirically it's absent on pure
        // content edits, but present on edits affecting either the stream
        // or topic:
        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60update_message.60.20event/near/1296823
```

And even if we did, this is a nice and intuitive conditional, even though I guess it would be kind of belt-and-suspenders, right.

Reply:
Updated that above comment to reflect the further discussion in that thread: we plan to make `stream_id` present on all updates to stream messages. Then this check will be directly needed in order to skip pure content edits.